jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketConnection.java

```diff
@@ -347,12 +348,10 @@ public boolean moreDemand()
 
             if (!fillingAndParsing)
                 throw new IllegalStateException();
-            if (demand > 0)
+            if (demand != 0)
```

Comment:
Add comment to say that cancelled demand creates synthentic demand to read the -1